src/WebViewShared.tsx

```diff
@@ -47,22 +64,38 @@ const createOnShouldStartLoadWithRequest = (
     lockIdentifier: number,
   ) => void,
   originWhitelist: readonly string[],
+  deepLinkWhitelist: readonly string[],
   onShouldStartLoadWithRequest?: OnShouldStartLoadWithRequest,
 ) => {
+  const compiledWhiteList = compileWhitelist(originWhitelist)
+
   return ({ nativeEvent }: ShouldStartLoadRequestEvent) => {
     let shouldStart = true;
     const { url, lockIdentifier, isTopFrame } = nativeEvent;
 
-    if (!_passesWhitelist(compileWhitelist(originWhitelist), url)) {
-      Linking.canOpenURL(url).then((supported) => {
-        if (supported && isTopFrame && /^https:\/\//.test(url)) {
-          return Linking.openURL(url);
+    /** Check if the url passes the origin whitelist */
+    if (!_passesWhitelist(compiledWhiteList, url)) {
+      /** Check if the url passes the hardcoded deeplink blocklist */
+      const foundMatchInBlocklist = matchWithRegexList(defaultDeeplinkBlocklist, url)
+      if (!foundMatchInBlocklist) {
+        /** Check if the url passes the dynamic deeplink allow list */
+        const foundMatchInAllowlist = matchWithPrefixStringList(deepLinkWhitelist, url)
```

Comment:
this implementation can fall through on `['']` being used as a whitelist

It should check something like `allowlist.includes(new URL(url).protocol)`

Reply:
Addressed in https://github.com/ExodusMovement/react-native-webview/pull/23